### PR TITLE
Movie site by Noemi and Daniela

### PR DESCRIPTION
**Pull request by dannuzak**
Link to my deployed project: 

https://movie-site-by-noemi-and-daniela.netlify.app/

My reflections on how this week's project turned out:
It was fun to do the pair programming and the project turned out well. We aimed for blue level.

Things I'd like to have clarified or explained in more detail:
The conditional rendering for not found.

### CODE REVIEW

File: code/src/index.css [hunk 1]
```diff
@@ -1,13 +1,189 @@
...
+	align-items: center;
+	width: 100%;
+	color: white;
+}
+
+.movie-poster {
+	width: 100%;
+	border: 2px solid white;
+	margin: 25px 0;
+}
```
Comment (MarySparrow) [inline code/src/index.css:103]: Good namings on the different classes, easy to find and understand!🤩

File: code/src/index.css [hunk 2]
```diff
@@ -1,13 +1,189 @@
...
+	color: red;
+	align-self: flex-start;
+	margin-top: 0;
+}
+
+@media (min-width: 768px) {
+	.movie-list-container {
+		display: flex;
+		flex-direction: row;
+		flex-wrap: wrap;
```
Comment (MarySparrow) [inline code/src/index.css:121]: Very good job on the details for the media queries! They are often forgotten, nice to see that you put some hard work into this ! Remember you do not have to repeat something that is already mentioned for mobile version, e.j display: flex . Anything that is changed is the need to overwrite it for the next media query. Saves many lines of code 👌

File: code/src/pages/MovieDetail.js [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+    fetch(MOVIE_DETAIL_URL)
+    .then((res) => res.json())
+    .then((json) => {
+      setMovieDetail(json)
+      console.log(json) 
+    })
+  }, [MOVIE_DETAIL_URL])
+
+  return (
+    <div> {movieDetail.length===0 ?<></> :
```
Comment (MarySparrow) [inline code/src/pages/MovieDetail.js:21]: We are not sure what this do, could you explain to us? *two lost sheeps* 🐑 🐑

File: code/src/pages/MovieDetail.js [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+      setMovieDetail(json)
+      console.log(json) 
+    })
+  }, [MOVIE_DETAIL_URL])
+
+  return (
+    <div> {movieDetail.length===0 ?<></> :    
+      <section className="movie-details-page" style={{backgroundImage:`url(https://image.tmdb.org/t/p/w1280/${movieDetail.backdrop_path})`}}>
+        <Link to={`/`} style={{ textDecoration: 'none' }}>
+          <div className="back-arrow-wrapper">
```
Comment (MarySparrow) [inline code/src/pages/MovieDetail.js:24]: Although we saw in the Technigo brief they use inline styling, they have also mentioned it is not the best practice. Maybe instead use a css for this details. Or maybe Karol could enlighten us of When this is suppose to be used 🤠 🤠

File: code/src/pages/MovieList.js [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+import { MOVIES_URL } from "reusable/urls"
+
+export const MovieList = () => {    
+  const [movies, setMovies] = useState([])
+
+  useEffect(() => {
+    fetch (MOVIES_URL)
+    .then(res => res.json())
+    .then(json => setMovies(json.results))  
+    .catch(err => console.error(err))
```
Comment (MarySparrow) [inline code/src/pages/MovieList.js:13]: might be a taste of ours but indentation could be added for these three lines when they are coming after a fetch() 😇

**Review by MarySparrow — APPROVED** [review]
Nice work on this project, 🍭 Good impression when watching the live page, works smoothly transitions to ipad, iphone etcetera. 😃  You seem to have put energy into it. ✨✨ Sorry you couldn't make the MovieThumb as you wanted.

File: code/src/pages/MovieDetail.js (hunk 3 above)
Comment (dannuzak) [reply code/src/pages/MovieDetail.js:21]: This prevents the page to go white (not found) when the user refreshes the page. ;)

**Review by KarolCislowski — CHANGES_REQUESTED** [review]
Nice project but please remove all console outputs.

To create 404 page you need to:
1. throw en error inside of `.then` block,
2. catch it in the `.catch` block and redirect it with `useHistory` hook provided by `React-router`
3. create a separate page(component) where the user will be redirected.
4. extra step if you are using netlify to host your app: create inside the `public` folder the `_redirects` file and save this line in it: `/* /index.html 200`

**Review by KarolCislowski — APPROVED** [review]

**Pull request closed without merging**